### PR TITLE
Merge 2.0 branch to master

### PR DESCRIPTION
**Pull request by BenWilson2**
## Related Issues/PRs

 #xxx

## What changes are proposed in this pull request?

MLflow 2.0 breaking changes. Please review commits to branch-2.0 to see a list of changes. 

## How is this patch tested?

- [ ] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

MLflow 2.0 breaking changes

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [X] `area/build`: Build and test infrastructure for MLflow
- [X] `area/docs`: MLflow documentation pages
- [X] `area/examples`: Example code
- [X] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [X] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [X] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [X] `area/projects`: MLproject format, project running backends
- [X] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [X] `area/server-infra`: MLflow Tracking server backend
- [X] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [X] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [X] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [X] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [X] `integrations/azure`: Azure and Azure ML integrations
- [X] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [X] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by BenWilson2** [comment]
@mlflow-automation autoformat

File: mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowClient.java [hunk 1]
```diff
@@ -348,6 +348,8 @@ public ExperimentsPage searchExperiments(String searchFilter,
     SearchExperiments.Response response = mapper.toSearchExperimentsResponse(ojson);
     return new ExperimentsPage(response.getExperimentsList(), response.getNextPageToken(),
       searchFilter, experimentViewType, maxResults, orderBy, this);
+<<<<<<< HEAD
+=======
```
Comment (harupy) [inline mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowClient.java:352]: Can we fix this?

File: mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowProtobufMapper.java [hunk 2]
```diff
@@ -186,14 +186,24 @@ GetExperiment.Response toGetExperimentResponse(String json) {
     return builder.build();
   }
 
+<<<<<<< HEAD
```
Comment (harupy) [inline mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowProtobufMapper.java:189]: Same here

File: mlflow/pipelines/utils/execution.py [hunk 3]
```diff
@@ -580,19 +580,31 @@ def __format__(self, path_spec: str) -> str:
 	cd {path:prp/} && \
         python -c "from mlflow.pipelines.steps.ingest import IngestScoringStep; IngestScoringStep.from_step_config_path(step_config_path='{path:exe/steps/ingest_scoring/conf.yaml}', pipeline_root='{path:prp/}').run(output_directory='{path:exe/steps/ingest_scoring/outputs}')"
 
+<<<<<<< HEAD
```
Comment (harupy) [inline mlflow/pipelines/utils/execution.py:583]: 👆

Comment (harupy) [inline mlflow/pipelines/utils/execution.py:583]: Same here

File: mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowClient.java (hunk 1 above)
Comment (BenWilson2) [reply mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowClient.java:352]: working on a huge list of fixes right now :)

**Comment by harupy** [comment]
@BenWilson2 Applied the only-latest label to avoid running all the cross-version tests.

**Pull request closed without merging**